faceV4.py

```diff
@@ -0,0 +1,202 @@
+        verify=False)
+        content = resp.content
+        data = json.loads(content)
+        return data
+
+#adds the name, timestamp, camera name, and sighting number to a csv
+def csv_add(value, csv_data, count, data_camera, baseURL, header):
+    timestamp = human_time(value['eventTimestamp'])
+    csv_data.append([])
+    name = value['faceName']
+    csv_data[count].append(name)
+    csv_data[count].append(timestamp)
+    camera = camera_name(value['deviceUuid'], data_camera)
+    csv_data[count].append(camera)
+    thumbnail = baseURL + value['thumbnailS3Key']
+    saving_img(thumbnail, name, count)
+    csv_data[count].append(count + 1)
+    with open(args.report + '/' + args.csv + '.csv', 'w', newline = '') as f:
+        writer = csv.writer(f)     # create the csv writer
+        writer.writerow(header)    # write the header
+        writer.writerows(csv_data) # write the data
+
+def csv_work(data_recentFaces):
+    count = 0
+    baseURL = "https://media.rhombussystems.com/media/faces?s3ObjectKey="
+    header = ['Name', 'Date', 'Camera', 'Sighting']
+    csv_data = []
+    data_camera = camera_data()
+    #gets a path and makes a directory file to the path
+    path = os.getcwd()
+    os.mkdir(path + '/' + args.report)
+    #checks if there is an arguement for a name to filter and only get instances with them
+    if args.name:
+        final_list = [event for event in data_recentFaces['faceEvents'] if event["faceName"] == args.name]
+        for value in final_list:
+            csv_add(value, csv_data, count, data_camera, baseURL, header)
+            count += 1
+    #checks for an arguement to filter only certain cameras
+    elif args.cameraName:
+        for value in data_camera['cameraStates']:
+            if args.cameraName == value['name']:
+                uuid = value['uuid']
+        final_list = [event for event in data_recentFaces['faceEvents'] if event["deviceUuid"] == uuid]
+        for value in final_list:
+            csv_add(value, csv_data, count, data_camera, baseURL, header)
+            count += 1
+    #checks for arguements of both certain cameras and a certain person to filter
+    elif args.cameraName and args.name:
+        for value in data_camera['cameraStates']:
+            if args.cameraName == value['name']:
+                uuid = value['uuid']
+        initial_list = [event for event in data_recentFaces['faceEvents'] if event["faceName"] == args.name]
+        final_list = [event for event in initial_list if event["deviceUuid"] == uuid]
+        for value in final_list:
+            csv_add(value, csv_data, count, data_camera, baseURL, header)
+            count += 1
+    #if no filter arguements it just grabs all from that time
+    else:
+        for value in data_recentFaces['faceEvents']:
+            csv_add(value, csv_data, count, data_camera, baseURL, header)
+            count += 1
+
+def main():
+    data_recentFaces = recent_faces()
+    csv_work(data_recentFaces)
+
+main()
```

Comment:
Add check to make sure that this file is being run directly, and not imported:

if __name__ == "__main__":
    main()